### ISSUE TITLE
Listing Changes Files Refresh Issue

### ISSUE THREAD

**Issue by drguthals** [github-for-unity/Unity#963]
[From Twitter](https://twitter.com/MaximumHussey/status/1062216585597792256):
```
I cannot for the life of me find a fix for the "Listing change files" refresh issue for the GitHub panel in Unity. How do I fix this refresh problem, I cannot submit my changes to my repos because it keeps deselecting my changes...

For context; this is what I see refreshing every second or so. It deselects my changes which prevents me from submitting them to the repo.
```
![dr3amjtvaaa4t9g](https://user-images.githubusercontent.com/1314285/48568615-a1234280-e8b4-11e8-9935-1193aef8fba6.jpg)

**Comment by shana** [comment]
Sorry that's it's so messed up for you! There's teo problems here, it seems: 1) it should not be constantly refreshing and 2) the UI should not reset your selection

I'll work on fixing 2, but I'm not surr why 1 is happening. Could you enable trace logging in the settings tab and then upload the log after some refreshes so I can take a look? The path to the log shows up in the console the first time you start Unity with the plugin running.

**Comment by MaximumHussey** [comment]
Absolutely! I'll see if I can get that for log you this afternoon when I get home.

Thanks for the assistance!

**Comment by MaximumHussey** [comment]
Hi again, sorry about the delay had a few busy days. I believe this is the correct log file.

Thanks again!

[github-unity.log](https://github.com/github-for-unity/Unity/files/2593239/github-unity.log)

**Comment by MaximumHussey** [comment]
To be honest I don't know, this is my first time using a GIT system before and I was hopin' this panel would make it easier to understand :P I'm super fresh to the whole scripting/programming thing!

I can't imagine that there would be anything causing it to refresh all the time. It seems that if I can press the "refresh" button at just the right time I am able to stop the panel from automatically doing it.

Is there anything else that I would be able to grab for you that might give you a clearer idea?

Thanks again!

**Comment by shana** [comment]
@MaximumHussey Sorry for not getting back to you earlier, still trying to figure out what's wrong. We're working on a fix for the selection, so at least even if it goes nuts refreshing it shouldn't affect your work. Hopefully I can get a build for you to test this week. Thank you for the patience!

**Comment by jeliastam** [comment]
I just experienced this issue.  Here is the log and I will try to provide any info upon request!

[github-unity.log](https://github.com/github-for-unity/Unity/files/2834460/github-unity.log)

**Comment by uberyoji** [comment]
I have the same issue once in a while... it is very annoying. To get out of that state, I usually close the editor and commit using Git Desktop... I hope it gets fixed soon. I have a feeling it is related to locks as I often trigger this loop state when working with locks.

**Comment by DanielDimanov** [comment]
I have the same issue also. What I do is I go to Window->Github CLI and as long as you are in the opened console/terminal it works fine. The UI Window does not work for me though. I hope it gets fixed soon!

**Comment by DankParrot** [comment]
Do you have any other git repo folders inside your project that aren't proper submodules? Fixing that fixed this issue for me.

**Comment by oddlots** [comment]
I have had this issue on MacOS 10.14.5 Running GitHubForUnity:'v1.4.0' Unity:'v2019.1.8f1'. To reiterate the issue, Checked items in the UI become unchecked by the refresh. 

The only things of interest in the trace log are:

> 190628-11:41:19.851 TRACE [ 1] <GitUserCache>                      needsInvalidation isInitialized:False timedOut:False
190628-11:41:19.853 TRACE [81] <ProcessWrapper>                    Running '/usr/local/bin/git config --get --global user.name'
190628-11:41:19.995 TRACE [85] <ProcessWrapper>                    Running '/usr/local/bin/git config --get --global user.email'
190628-11:41:20.063 TRACE [73] <ProcessWrapper>                    Running '/usr/local/bin/git -c i18n.logoutputencoding=utf8 -c core.quotepath=false --no-optional-locks status -b -u --porcelain'

It repeats the GitStatusCache line occasionally and ProcessWrapper line quite a bit. 

- I made sure that the repo was in sync and the "listing file changes" progress bar kept up its antics. 
- I tried a completely blank Unity project and initialized a repo from the plugin and it did **not** have the issue.
- I returned to the my existing project and synced up the gitignore and gitattributes files. issue remained.

**Comment by Snoozey** [comment]
Stuck with the same issue. 

Any one have a solution?

**Comment by oddlots** [comment]
Unfortunately I abandoned it after a number of attempts

**Comment by Bosspwn** [comment]
Same issue.
Wandering how to deal with it.

**Comment by Bosspwn** [comment]
Same issue.
Enabled trace logging and found a piece of logs that have been repeated
><ProcessWrapper> Running '/usr/bin/git -c i18n.logoutputencoding=utf8 -c core.quotepath=false --no-optional-locks status -b -u --porcelain'

Wondering how to fix it.

**Comment by kronnect** [comment]
I have also this issue.

**Comment by sarrowsmith** [comment]
Editor->Project Settings->Editor: Version Control/Mode = Visible Meta Files
fixed it for me.

**Comment by douyafang** [comment]
Same issue, sarrowsmith's solution fixed the problem for me, thanks.

**Comment by itsandrewhong** [comment]
sarrowsmith's solution worked for me too! Thanks!

**Comment by Crazykingjammy** [comment]
wow that worked for me too. Anyone here trying to be foolish and have both collab and git repo activated?

**Comment by toonhimself** [comment]
This solution also worked for me!

**Comment by IONM12** [comment]
github is useless, what a lamer programmer

**Comment by fweibel** [comment]
Changing the visibility mode of meta files doesn't seems to work for me. I am using Unity 2021.2.0b11,which is a beta version.
The refresh bug is reproduced everytime I create a project and initialize Git. It might be due to a change in unity.

**Comment by smultar** [comment]
> Changing the visibility mode of metafiles doesn't seem to work for me. I am using Unity 2021.2.0b11, which is a beta version. The refresh bug is reproduced every time I create a project and initialize Git. It might be due to a change in unity.

Any luck on fixing this issue?

**Comment by GusDS** [comment]
Today I upgraded to the new 2021.2 "Tech Stream", and the "Listing changed files" is happening now with my project.
The "Visible Meta Files" fix above doesn't work anymore.
Any solution for this 3 years old issue?

**Comment by Kazbek** [comment]
> Today I upgraded to the new 2021.2 "Tech Stream", and the "Listing changed files" is happening now with my project. The "Visible Meta Files" fix above doesn't work anymore. Any solution for this 3 years old issue?

Have same question. Mode already "Visible Meta Files".

**Comment by x1rom** [comment]
So changing the Mode did not do anything for me, it was already in the right mode, but i have [this file](https://github.com/github/gitignore/blob/master/Unity.gitignore) lying around in my root directory. And renaming the gitignore file solved the issue for me. But obviously i need the file.

Edit: False Alarm, it just takes a lot of time for it to refresh, but it still does it.

**Comment by someone-s** [comment]
my temporary solution is to use github-desktop app

**Comment by GusDS** [comment]
> my temporary solution is to use github-desktop app

Yes. When trying to fix this, I did a logout and couldn't login again... This is ANOTHER posted issue without a solution.
I've already dropped GitHub for Unity and I'm using GitHub desktop or Visual Studio Code source control.
The last release of this package is from June 2019. It seems it's discontinued and without support.

**Comment by cpizano** [comment]
Got hit by this issue today w/ unity 2021.2.7f1 on Windows 10. Same thing it keeps running git status every 800 ms and de-selecting the files to commit. Downloaded github-for-unity today and this is a brand new unity project.  The "visible meta files" is already set and switching it does nothing.

**Comment by CoolCreasu** [comment]
Hello, I seem to have this issue too, Is this already supposed to be fixed or not? I can't get my project on GitHub with the plugin.
It always keeps repeating the Listing changed files thing and also resets the selected files to unselected.

**Comment by Mushe94** [comment]
We are getting the same issue with Unity 2021.2.7f1. We only use GitHub for Unity to see modified assets and lock assets, the rest is used through GitHub Desktop.

**Comment by intermension** [comment]
Same issue for me with 2021.3.1f1 in May 2022. 

I downloaded the source code and it seems the issue is in the ProcessTask wrapper for GitStatusTask.cs not returning proper status /  possibly throwing which in turn causes the ApplicationCache to continually re-invalidate. 

Either way it  appears this repo has been abandoned with no updates since 2019 so I have removed it completely by deleting the GitHub folder from project Assets. It should probably be removed from GitHub links since it doesn't appear to work at all and is not being maintained. 

I am just using [GitHub Desktop](https://git-scm.com/downloads/guis) now, which you can easily configure to also use [Git Large File Storage](https://docs.github.com/en/repositories/working-with-files/managing-large-files/configuring-git-large-file-storage). Takes 5mins to setup.

**Comment by PuppedToy** [comment]
Same issue here with 2021.2.8f1

**Comment by Mushe94** [comment]
> Either way it appears this repo has been abandoned with no updates since 2019 so I have removed it completely by deleting the GitHub folder from project Assets. It should probably be removed from GitHub links since it doesn't appear to work at all and is not being maintained.

I wouldn't remove it since it's very helpful to highlight modified assets, only because of that feature the package is insanely worth it. The other one as a "nice to have" is locking files. And that's it, the rest you can handle using GitHub Desktop like you said. They really only need to fix this constant refreshing issue.

**Comment by bigwitch3r** [comment]
2021.3.3f1. Only GitHub Desktop helps me.

**Comment by jlopezr** [comment]
One thing that worked for me is checking the git and git-lfs paths. If yours is something like C:\Program Files\Git\git-lfs.exe, add quotation marks around the name, i.e "C:\Program Files\Git\git-lfs.exe". Seems that Github-for-Unity does not like path with spaces in the middle, because it tries to use c:\program as the git and git-lfs executables.

**Comment by CoolCreasu** [comment]
> One thing that worked for me is checking the git and git-lfs paths. If yours is something like C:\Program Files\Git\git-lfs.exe, add quotation marks around the name, i.e "C:\Program Files\Git\git-lfs.exe". Seems that Github-for-Unity does not like path with spaces in the middle, because it tries to use c:\program as the git and git-lfs executables.

Hello where exactly do you change this in which files?

**Comment by btbxbob** [comment]
> One thing that worked for me is checking the git and git-lfs paths. If yours is something like C:\Program Files\Git\git-lfs.exe, add quotation marks around the name, i.e "C:\Program Files\Git\git-lfs.exe". Seems that Github-for-Unity does not like path with spaces in the middle, because it tries to use c:\program as the git and git-lfs executables.

Hello, me too, where is this config file please?

**Comment by Mushe94** [comment]
@CoolCreasu @btbxbob 

Window - GitHub, add the " like this:

![GitHub for Unity](https://user-images.githubusercontent.com/40760950/183474715-52ce9e23-8545-418a-a299-c07614e0afd8.png)


Honestly for me it didn't work at all, it keeps refreshing.

**Comment by jpalczewski** [comment]
And this is still an issue for me on Windows 11 and unity 2021.3.11f1. Nothing mentioned in this issue have helped.

**Comment by shana** [comment]
TL;DR: **PLEASE** for the love of $deity go to https://github.com/spoiledcat/git-for-unity and install Git for Unity from the instructions there (remove GitHub for Unity from your project first)

Hey everyone :wave:

As you have probably noticed, this project is dead. I know it's very frustrating to have a tool that used to work and is now bitrotting. I'm sorry about that, and I'm not sure what I can do with this particular repo, since it's not... exactly... mine (it's complicated). I forked this for Unity back in 2019 - which is called Git for Unity - and in the process I fixed a metric ton of issues with it, probably including this issue. That fork, of course, haha, is also not maintained anymore by Unity, so it's also bitrotting.

I don't have a lot of time, but I've decided to put Git for Unity on a permanent home for myself, and resume work on it, at https://github.com/spoiledcat/git-for-unity

For now, to solve your immediate problem, replace the old GitHub for Unity with the latest Git for Unity package that I have available from my registry. Instructions on how to install it are at https://github.com/spoiledcat/git-for-unity. 

If you want to be notified when I update the packages and put out new builds, head over to https://github.com/spoiledcat/git-for-unity/issues/1 and subscribe to that issue.

If there are issues, open them at https://github.com/spoiledcat/git-for-unity/issues.

Cheers,
shana

**Comment by sergiJimenez** [comment]
Hello guys,
I already "fix" the problem. I mean "fix" because it only stops the "listing changes files" refresh but you can't upload the project to GitHub. When you installed the Github for Unity it will created a new folder in your Unity project. It called "Plugins". Inside this folder you will find another folder calls "GitHub". I removed the folder "Plugins" and then the "listing changes files" refresh stops. Choosing this way I found a solution for my issue and maybe helps you too. Now I can do a version control using GitHub Desktop instead of GitHub for Unity. I hope this information helps you too!

Keeps always programming and always smiling!

**Comment by Mushe94** [comment]
> Hello guys, I already "fix" the problem. I mean "fix" because it only stops the "listing changes files" refresh but you can't upload the project to GitHub. When you installed the Github for Unity it will created a new folder in your Unity project. It called "Plugins". Inside this folder you will find another folder calls "GitHub". I removed the folder "Plugins" and then the "listing changes files" refresh stops. Choosing this way I found a solution for my issue and maybe helps you too. Now I can do a version control using GitHub Desktop instead of GitHub for Unity. I hope this information helps you too!
> 
> Keeps always programming and always smiling!


So your solution for the issue is to delete the tool? 🤣

**Comment by sergiJimenez** [comment]
> > Hello guys, I already "fix" the problem. I mean "fix" because it only stops the "listing changes files" refresh but you can't upload the project to GitHub. When you installed the Github for Unity it will created a new folder in your Unity project. It called "Plugins". Inside this folder you will find another folder calls "GitHub". I removed the folder "Plugins" and then the "listing changes files" refresh stops. Choosing this way I found a solution for my issue and maybe helps you too. Now I can do a version control using GitHub Desktop instead of GitHub for Unity. I hope this information helps you too!
> > Keeps always programming and always smiling!
> 
> So your solution for the issue is to delete the tool? 🤣

Well, it's not a definetly solution for the issue. It's only a solution to stop "listing changes files" like I said in the reply. Maybe, someone like me, installed the plugin and he/she found the issue when it's too late. But, until the problem will be solve, I suggest to do this (https://www.youtube.com/watch?v=sjlVox4f5Ww), but before unistall the plugin. In my case works perfect. Maybe it's a little bit annoying to open GitHub Desktop each time that you want to use Unity to push the commit's but you don't have to push the commit's in the moment to save it. I hope this information helps you too!

Keeps always programming and always smiling!

**Comment by Mushe94** [comment]
> > > Hello guys, I already "fix" the problem. I mean "fix" because it only stops the "listing changes files" refresh but you can't upload the project to GitHub. When you installed the Github for Unity it will created a new folder in your Unity project. It called "Plugins". Inside this folder you will find another folder calls "GitHub". I removed the folder "Plugins" and then the "listing changes files" refresh stops. Choosing this way I found a solution for my issue and maybe helps you too. Now I can do a version control using GitHub Desktop instead of GitHub for Unity. I hope this information helps you too!
> > > Keeps always programming and always smiling!
> > 
> > 
> > So your solution for the issue is to delete the tool? 🤣
> 
> Well, it's not a definetly solution for the issue. It's only a solution to stop "listing changes files" like I said in the reply. Maybe, someone like me, installed the plugin and he/she found the issue when it's too late. But, until the problem will be solve, I suggest to do this (https://www.youtube.com/watch?v=sjlVox4f5Ww), but before unistall the plugin. In my case works perfect. Maybe it's a little bit annoying to open GitHub Desktop each time that you want to use Unity to push the commit's but you don't have to push the commit's in the moment to save it. I hope this information helps you too!
> 
> Keeps always programming and always smiling!


The real utility of the plugging isn't to push and pull inside of Unity, because using GitHub Desktop is faster and has more tools. The utility is too see inside the editor what things you've changed and created.

**Comment by vltmedia** [comment]
If anyone hits this around 4/19/2023 and using around Unity2022.1.23f here's a **fix**. 
1. Uninstall this package.
2. Install this newer Github Unity implementation.
[Github - Unity](https://github.com/github-for-unity/Unity)

No more annoying GUI refreshing!
